components/omega/doc/design/SubmesoscaleEddies.md

```diff
@@ -0,0 +1,345 @@
+        SubmesoEddy(const HorzMesh *Mesh, const VertCoord *VCoord, Config *Options); // Class constructor
+
+    private:
+        // Mesh dimensions
+        I4 NVertLayers;
+        I4 NVertLayersP1;
+        I4 NEdgesOwned;
+        I4 NEdgesAll;
+        I4 NEdgesSize;
+
+        // Vertical loop bounds from VertCoord
+        Array1DI4 MinLayerEdgeBot;
+        Array1DI4 MaxLayerEdgeTop;
+
+        // Arrays from HorzMesh
+        Array2DI4 CellsOnEdge;
+        Array2DReal EdgeSignOnCell;
+
+        Array2DReal SubmesoStreamFunc;
+        Array1DReal BuoyGradML;
+        Array1DReal TimeScale;
+}
+```
+
+### 4.2 Methods
+
+#### 4.2.1 Class Constructor
+
+The constructor will store necessary static mesh information as private variables, handle config options, and compute fixed quantities like the spatially varying $\tau$ variable.
+```c++
+SubmesoEddy::SubmesoEddy(const HorzMesh *Mesh, const VertCoord *VCoord, Config *Options)
+```
+
+#### 4.2.1 Initialization
+
+The initialization method will read the config parameters and register necessary fields for the calculation and diagnostic output.
+```c++
+int SubmesoEddy::init(Config *Options)
+```
+
+#### 4.2.2 Density MLD calculation
+
+The public `computeDenMLD` mthod will utilize the specific volume frmo the TEOS-10 calculation to compute a mixed layer depthy based on the threshold criterion. Given Omega has easy access to `SpecVol` and not density, we rearrange the standard density threshold calculation
+$$
+\rho(x,y,z,t) - \rho(x,y,10,t) \geq 0.03
+$$
+
+as
+
+$$
+\frac{1}{\alpha(x,y,z,t)} - \frac{1}{\alpha(x,y,10,t)} \geq 0.03.
+$$
+
+This can be rearranged to
+
+$$
+\frac{\alpha(x,y,10,t) - \alpha(x,y,z,t)}{\alpha(x,y,z,t)} = \frac{\alpha(x,y,10,t)}{\alpha(x,y,z,t)} - 1 \geq 0.03 \alpha(x,y,10,t)
+$$ (mld-defn)
+
+This form allows for easier use of the `SpecVol` variable in Omega  with minimal conversions in the calculation.  The specific volumes in the calculation are referenced to pressure at 10m.  The fundamental `SpecVol` calculation in Omega is
+
+```c++
+SpecVol(ICell, K) =
+    calcRefProfile(Pressure(ICell, K)) +
+    calcDelta(LocSpecVolPCoeffs, KVec, Pressure(ICell, K));
+```
+
+Here, the computation of `LocSpecVolPCoeffs` is the most computationally expensive (the 75-term polynomial).  This calculation is only dependent on $\Theta$ and $S$ and can thus be reused for any displacement case.  With calculation of a `SpecVol10mDisplaced` variable, calculation of the mixed layer depth follows from [](#mld-defn).  If the MLD falls between two model layers, linear interpolation will be used to find the final MLD.
+
+```c++
+void computeDenMLD(Array1DReal &DenMLD, const Array2DReal &SpecVol, const Array2DReal &ZInterface)
+```
+
+#### 4.2.3 Buoyancy gradient calculation
+
+For Omega, we will define the buoyancy as
+
+$$
+b = g \frac{\rho - \rho_0}{\rho}
+$$
+
+where $\rho_0$ is the reference density used in the pseudo height coordinate definition.  This can be easily rearranged to
```

Comment:
```suggestion
where $\rho_0$ is the reference density used in the pseudo height coordinate definition.  Again, since Omega calculates `SpecVol` and not density itself, this can be easily rearranged to
```